### ISSUE TITLE
sgd - optimal learning rate with line search

### ISSUE THREAD

**Issue by erodola** [erodola/DLAI-s2-2021#9]
Let me clarify the confusion on the orthogonal directions obtained with the optimal alpha in slide 66/127 of the lecture "Stochastic gradient descent".

Please read this while looking at the slide.

- Let's say we are at the point **x_t** (first red dot in the figure). 
- The optimal alpha minimizes (globally) the function **f** along the direction **-grad(f(x_t))**.
- Let's call the optimal point **x*** (second red dot).
- Since **x*** is a minimum, the directional derivative of **f(x*)** along direction **-grad(f(x_t))** is equal to zero. 
- By definition, this directional derivative can be written as the inner product between **grad(f(x*))** and the direction **-grad(f(x_t))**.
- Having this inner product equal to zero means that **grad(f(x*))** and **-grad(f(x_t))** are orthogonal. 
- Since **grad(f(x*))** is also the direction of the next iteration of gradient descent, we get that successive directions of gradient descent with line search are always orthogonal.

**Comment by ShakirKhurshid** [comment]
**y is minimum wrt what?** isn't  the point in the centre of the function (blue area) the minimum of the function
and **what is the direction of grad(f(y))   ??** 
can you please explain point 4, if you could draw out the directions? It would be really helpful.

**Comment by erodola** [comment]
I made the attached image to try clarifying the points you raised. Can you check if this works for you now? Please let me know if you have additional questions.

![sgdlinesearch](https://user-images.githubusercontent.com/9395545/115530142-cc839600-a293-11eb-836c-ecf080043d1b.png)
